src/async/serializeAsync.ts

```diff
@@ -264,6 +264,12 @@ export function createTsonSSEResponse(opts: TsonAsyncOptions) {
 				controller.enqueue(`data: ${JSON.stringify(chunk)}\n\n`);
 			}
 
+			// indicate the end of the stream
+
+			controller.enqueue(`data: ["__tson_disconnect"]\n\n`);
```

Comment:
A bit but could this be made into an exported constant which is imported where we depend on it?

Reply:
I'm also unsure about the things I pointed out on Discord

When we have an infinite stream, it might be useful to notify clients that we want them to reconnect without destroying the *existing* stream similarly to how we deal with it in subscriptions

Also unsure about the passing of a cursor when continuing SSE & if that's something we should consider making good abstractions for in trpc subscriptions.

This all said, it's a non issue for now. We can aim to get streamed responses done before we consider subscriptions too much.

I'm also leaning towards only doing SSE now that you've shown it can be supported. One downside is that EventSource needs a poly fill in node.

Reply:
> When we have an infinite stream, it might be useful to notify clients that we want them to reconnect without
> destroying the existing stream similarly to how we deal with it in subscriptions

Could you elaborate on that? I'm not sure I follow.

> Also unsure about the passing of a cursor when continuing SSE & if that's something we should consider making good
> abstractions for in trpc subscriptions.

I agree. It's not a feature of tRPC subscriptions (websockets) currently, but that's not to say it wouldn't be a
nice-to-have. I think SSE gives us a nice freebie out-of-the-box as far as connection integrity, which opens up a lot of
use-cases for finite streams. Infinite streams, resumable streams, cursor-based pagination, and other more complex
use-cases can be debated in a second pass (perhaps after some RFC).

> I'm also leaning towards only doing SSE now that you've shown it can be supported.

In what context? tSON? tRPC? Stream queries? Subscriptions? (I'm differentiating between short-lived, finite responses
and long-lived, potentially infinite responses... but the more I do so, the less sense it seems to make to me).

I don't mind a poly-fill so much. I would even go so far as to say we should ship the standard, pure-javascript polyfill
linked in the MDN docs with whatever package needs it (be it tRPC or some adapter)

Reply:
Let me know if you like the custom event approach I just pushed, or if you prefer it as a constant

Reply:
Custom event sounds cool

> In what context? tSON? tRPC? Stream queries? Subscriptions? (I'm differentiating between short-lived, finite responses


Yeah I don't see the big reason why tRPC would use a different communication thingie for finite vs infinite - SSE seems to work for all [except Node]

> Let me know if you like the custom event approach I just pushed, or if you prefer it as a constant

It's cool, ship it